lib/src/qr_code.dart

```diff
@@ -10,27 +12,66 @@ import 'rs_block.dart';
 import 'util.dart' as qr_util;
 
 class QrCode {
-  final int typeNumber;
+  int _typeNumber;
   final int errorCorrectLevel;
-  final int moduleCount;
+  int _moduleCount;
   final List<List<bool>> _modules;
   List<int> _dataCache;
   final List<QrByte> _dataList = <QrByte>[];
 
-  QrCode(this.typeNumber, this.errorCorrectLevel)
-      : moduleCount = typeNumber * 4 + 17,
+  int get typeNumber => _typeNumber;
+
+  int get moduleCount => _moduleCount;
+
+  QrCode(this._typeNumber, this.errorCorrectLevel)
+      : _moduleCount = _typeNumber * 4 + 17,
         _modules = <List<bool>>[] {
-    RangeError.checkValueInInterval(typeNumber, 1, 40, 'typeNumber');
+    _init();
+  }
+
+  QrCode.fromData({@required String data, @required this.errorCorrectLevel})
```

Comment:
Could you do this as a factory? Then yo wouldn't have to make typeNumber & moduleCount non-final.

Reply:
done